deployer/src/persistence/mod.rs

```diff
@@ -459,13 +496,79 @@ impl ResourceManager for Persistence {
                     .expect("to have the resource recorder set up")
                     .get_service_resources(service_resources_req)
                     .await
-                    .map_err(Error::from)
+                    .map_err(PersistenceError::ResourceRecorder)
                     .map(|res| res.into_inner());
             }
         }
 
         Ok(res)
     }
+
+    async fn get_resource(
+        &mut self,
+        service_id: &Ulid,
+        r#type: String,
```

Comment:
Same here, this should be the resource enum to reduce errors

Reply:
Done.